### PR TITLE
Restructure API docs to use better type signatures

### PR DESCRIPTION
**Pull request by bjz**
I was having a great deal of trouble understanding the API, so I've attempted to make the type signatures much clearer in the docs. I may have got some things wrong though. I originally stuck to the Haskell-esque syntax for the type signatures, but opted to switch to TypeScript syntax because it allows you to annotate the function parameters. I hope this helps!

[Rendered](https://github.com/bjz/react-motion/blob/d072fa928baf5a5955b9ae1bc182e111377ca791/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,25 +52,67 @@ For 95% of use-cases of animating components, we don't have to resort to using h
...
+
+### `class Motion extends React.Component`
+
+#### `props`
+
+##### `defaultStyle?: Style`
 Optional. The value when the component first renders (ignored in subsequent renders). Accepts an object of arbitrary keys, mapping to initial values you want to animate, e.g. `{x: 0, y: 10}`.
 
-#### `style: Object`
+##### `style: Style`
```
Comment (nkbt) [inline README.md]: :+1: nice

**Comment by nkbt** [comment]
We are now moving to Flow code annotations, so it would make much more sense to use those for docs too instead of TypeScript

See #187

**Comment by nkbt** [comment]
In general LGTM, TS-->Flow annotations should be a quick fix

**Comment by bjz** [comment]
Oh cool, that is most excellent. Can you gnerate docs from Flow, like with typescript?

**Comment by bjz** [comment]
I'll amend with the Flow syntax, but it would be even better if they were auto-generated.

**Comment by nkbt** [comment]
I don't think there was any intention to auto-generate docs from Flow. It does not really have any context similar to what JSDoc does (via comments).

**Comment by bjz** [comment]
Ah, that's a shame. You can get very nice dos with little effort using generators like [TypeDoc](http://typedoc.io/), automatically generating hotlinked type signatures.

**Comment by bjz** [comment]
Wasn't much of a change to use flow-style annotations. Just meant I could use `?React.Component` as opposed ti `(React.Component | void)`

**Comment by nkbt** [comment]
:+1: LGTM, @chenglou there are some noticeable text changes, do they capture everything right?

**Comment by bjz** [comment]
Any updates on this?

**Comment by chenglou** [comment]
(Sorry catching up to like, everything... Will check more closely tonight)

File: README.md [hunk 2]
```diff
@@ -52,25 +52,67 @@ For 95% of use-cases of animating components, we don't have to resort to using h
 This library also provides an alternative, more powerful API for React's `TransitionGroup`.
 
 ## API
-The library exports `Motion`, `StaggeredMotion`, `TransitionMotion`, `presets`, `spring` and `utils`.
 
-#### `spring: number -> ?[stiffness, damping] -> ConfigurationObject`
-(**Note**: not the `Spring` component in version <0.3.0.)
-The pervasive helper used to specify the how to animate to the destination value, e.g. `spring(10, [120, 17])` returns an opaque configuration that describes "an animation to the value 10, with a physics spring's stiffness of 120 and damping of 17". `spring(10)` without the spring configuration array defaults to `[170, 26]`. See below for more usage and see [here](#presets) for a list of convenient configurations the library exports.
+### `type PhysicsConfig = [number, number]`
```
Comment (chenglou) [inline README.md]: Really not sure about this. If you check the final effect: https://github.com/bjz/react-motion/blob/restructure-readme/README.md#api it's a lot of ceremony before some actual docs

File: README.md [hunk 3]
```diff
@@ -52,25 +52,67 @@ For 95% of use-cases of animating components, we don't have to resort to using h
...
+
+```jsx
+let config = spring(10, [120, 17]);
+```
+
+#### Note
+
+This is not the `SpringConfig` component in version <0.3.0 - this has been renamed to `Motion` (see below).
+
+### `class Motion extends React.Component`
```
Comment (chenglou) [inline README.md]: `<Motion />` was much clearer

File: README.md [hunk 4]
```diff
@@ -213,10 +262,22 @@ const Demo = React.createClass({
 });
 ```
 
-### `presets`
+### Modules
+
+#### `presets`
```
Comment (chenglou) [inline README.md]: Not sure about this. Seems like the link's helpful enough.

**Comment by chenglou** [comment]
We're using Flow's type syntax. Also this is getting hard to read: 
screenshot 2015-10-14 01 27 49

The vague type annotations, where it's the case, are mostly intentional. I've debated long for/against this but ultimately went with a general type + a concise description. Formalizing things like this makes everything much more rigorous but actually decreases readability imo...

**Comment by chenglou** [comment]
Hey @bjz, this is really late, but hope you like the [new docs](https://github.com/chenglou/react-motion/blob/6a7a0f3f72610c6a7fdad2920b44e728ca6e172d/README.md#api)! =) 

Proper Flow type signature, with an exposed public types file.

Sorry that this took so long. To be fair, Flow wasn't (and is still a bit buggy sometimes) ready back then either. This PR has diverged too much so I'll close it. But thanks a lot for the initiative!

**Comment by bjz** [comment]
No worries - it looks great now!

**Pull request closed without merging**